app/components/audio-capturer.js

```diff
@@ -1,27 +1,51 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
+import { Plugins } from '@capacitor/core';
+const { VoiceRecorder, Device } = Plugins;
+import { Base64Binary } from 'wuf/utils/base64Binary';
 
 export default class AudioCapturerComponent extends Component {
   @action
   async startRecording() {
-    const stream = await navigator.mediaDevices.getUserMedia({
-      audio: true,
-      video: false
-    });
-    const options = { mimeType: 'audio/webm' };
-    this.mediaRecorder = new MediaRecorder(stream, options);
+    const deviceInfo = await Device.getInfo();
 
-    this.mediaRecorder.addEventListener('dataavailable', e => {
-      this.args.uploadAudioVideo({ blob: e.data });
-    });
+    if (deviceInfo.operatingSystem === 'ios') {
+      const canRecord = await VoiceRecorder.canDeviceVoiceRecord();
+      if (canRecord) {
+        const permissionGranted = await VoiceRecorder.requestAudioRecordingPermission();
+        if (permissionGranted) {
+          await VoiceRecorder.startRecording();
+        }
+      }
+    } else {
+      const stream = await navigator.mediaDevices.getUserMedia({
+        audio: true,
+        video: false
+      });
+      const options = { mimeType: 'audio/webm' };
+      this.mediaRecorder = new MediaRecorder(stream, options);
 
-    this.mediaRecorder.start();
+      this.mediaRecorder.addEventListener('dataavailable', e => {
+        this.args.uploadAudioVideo({ blob: e.data });
+      });
+
+      this.mediaRecorder.start();
+    }
   }
 
   @action
-  stopRecording() {
-    if (this.mediaRecorder) {
-      this.mediaRecorder.stop();
+  async stopRecording() {
+    const deviceInfo = await Device.getInfo();
+
+    if (deviceInfo.operatingSystem === 'ios') {
+      let result = await VoiceRecorder.stopRecording();
+      let byteArray = Base64Binary.decodeArrayBuffer(result.value.recordDataBase64);
+
+      this.args.uploadAudioVideo(byteArray);
```

Comment:
@jaredgalanis shouldn't this match the format from before where it's an object with `blob` as the key? `this.args.uploadAudioVideo({ blob: e.data });`

I think if we pass the same data format, we can ensure that it *should* draw the same.

Reply:
So I tried passing instantiating a Blob with the ArrayBuffer that we get out of converting the base64 string with this in mind, but it didn't work unfortunately.